gptme/tools/restart.py

```diff
@@ -33,25 +51,61 @@ def _do_restart(conversation_name: str | None = None):
     """
     import atexit
 
-    # Build restart command with explicit conversation name
+    # Build restart command, filtering out positional arguments (prompts)
+    # since they are already in the conversation log (Issue #1011)
     # Use sys.argv[0] (gptme script) not sys.executable (python interpreter)
-    restart_args = sys.argv[:]
+    filtered_args = [sys.argv[0]]  # Keep script name
+    skip_next = False
+    i = 1
 
-    # Ensure we have the conversation name in the args
-    if conversation_name:
-        # Remove any existing --name or --resume args and their values
-        filtered_args = []
-        skip_next = False
-        for arg in restart_args:
-            if skip_next:
-                skip_next = False
-                continue
-            if arg in ["--name", "-r", "--resume"]:
-                skip_next = True  # Skip this flag and the next argument (its value)
-                continue
+    skip_value = False  # True when we need to skip the next value (for --name)
+    while i < len(sys.argv):
+        arg = sys.argv[i]
+
+        if skip_next:
+            # This arg is a value for the previous flag, keep it
             filtered_args.append(arg)
-        restart_args = filtered_args
+            skip_next = False
+            i += 1
+            continue
+
+        if skip_value:
+            # This arg is a value for --name/--resume, skip it
+            skip_value = False
+            i += 1
+            continue
+
+        if arg.startswith("-"):
+            # This is a flag
+            if arg in ["--name", "-r", "--resume", "-m", "--model"]:
```

Comment:
Should also include `tools` and other things that get persisted to the chat config (e.g. stream, interactive, agent, workspace, tool format, etc)